docs/src/pages/autodiff.md

```diff
@@ -0,0 +1,152 @@
+```
+        x        Input
+       / \
+      /   \
+     u     v     Intermediate values computed from x
+      \   /
+       \ /
+        w        w = g(u, v)
+        |
+        y        Output
+```
+where control flow goes from top to bottom.
+
+To model the computational graph of a function, Nabla uses what it calls `Node`s, and it
+stores values to a `Tape`.
+`Node` is an abstract type with subtypes `Leaf` and `Branch`.
+A `Leaf` is a static quantity that wraps an input value and a tape.
+As its name suggests, it represents a leaf in the computational graph.
+A `Branch` is the result of a function call which has been "intercepted" by Nabla, in
+the sense that one or more arguments passed to it is a `Node`.
+It holds the value of from evaluating the call, as well as information about its position
+in the computational graph.
+Functions which should produce `Branch`es in the computational graph are explicitly
+extended to do so; this does not happen automatically for each function.
+
+### Forward pass
+
+Nabla starts `∇(f)(x)` off by creating a `Tape` to hold values and constructing a
+`Leaf` that references the tape and the input `x`.
+It then performs what's called the _forward pass_, where it executes `f` as usual, walking
+the computational graph from top to bottom, but with the aforementioned `Leaf` in place of
+`x`.
+As `f` is executing, each intercepted function call writes a `Branch` to the tape.
+The end result is a fully populated tape that will be used in the _reverse pass_.
+
+### Reverse pass
+
+During the reverse pass, we make another pass over the computational graph of `f`, but
+instead of going from top to bottom, we're working our way from bottom to top.
+
+We start with an empty tape the same length as the one populated in the forward pass,
+but with a 1 in the last place, corresponding to the identity ``\bar{y} = 1``.
+We then traverse the forward tape, compute the sensitivity for each `Branch`, and store
+it in the corresponding position in the reverse tape.
+This process happens in an internal function called `propagate`.
+
+The computational graph in question may not be linear, which means we may end up needing
+to "update" a value we've already stored to the tape.
+By the chain rule, this is just a simple sum of the existing value on the tape with the
+new value.
+
+### Computing derivatives
+
+As we're propagating sensitivities up the graph during the reverse pass, we're calling
+`∇` on each intermediate computation.
+In the case of `f`, this means that when computing the sensitivity `w̄` for the intermediate
+variable `w`, we will call `∇` on `g`.
+
+This is where the real power of Nabla comes into play.
+In Julia, every function has its own type, which permits defining methods that dispatch
+on the particular function passed to it.
+`∇` makes heavy use of this; each custom sensitivity is implemented as a method of `∇`.
+If no specific method for a particular function has been defined, Nabla computes the
```

Comment:
I'm not sure that this is quite correct. The `fmad` implementation is only really used inside `broadcast`. If a particular method of a function doesn't have a sensitivity defined for it, it simply enters the function and tracks the operations inside it.

Reply:
Oh, you're right. Whoops.